spec/dynamoid/persistence_spec.rb

```diff
@@ -292,6 +292,32 @@
         msg.destroy
       end.to_not raise_error
     end
+
+    context 'with lock version' do
+      it 'deletes a record if lock version matches' do
+        address.save!
+        expect { address.destroy }.to_not raise_error
```

Comment:
If you're expecting an exception, don't you have to use `#destroy!` instead of the kinder `#destroy`?